Sources/AirbnbSwiftFormatTool/AirbnbSwiftFormatTool.swift

```diff
@@ -40,40 +40,74 @@ struct AirbnbSwiftFormatTool: ParsableCommand {
   @Option(help: "The project's minimum Swift version")
   var swiftVersion: String?
 
-  mutating func run() throws {
-    try swiftFormat.run()
-    swiftFormat.waitUntilExit()
-
-    try swiftLint.run()
-    swiftLint.waitUntilExit()
-
-    if log {
-      log(swiftFormat.shellCommand)
-      log(swiftLint.shellCommand)
-      log("SwiftFormat ended with exit code \(swiftFormat.terminationStatus)")
-      log("SwiftLint ended with exit code \(swiftLint.terminationStatus)")
+  func run() throws {
+    let swiftFormat = makeSwiftFormatCommand()
+    let swiftLint = makeSwiftLintCommand(autocorrect: false)
+    let swiftLintAutocorrect = makeSwiftLintCommand(autocorrect: true)
+
+    let swiftFormatExitCode = try swiftFormat.run()
+
+    // We always have to run SwiftLint in lint-only mode at least once,
+    // because when in autocorrect mode SwiftLint won't emit any lint warnings.
+    let swiftLintExitCode = try swiftLint.run()
+
+    let swiftLintAutocorrectExitCode: Int32?
+    if
+      // When only linting, we shouldn't run SwiftLint with autocorrect enabled
+      !lintOnly,
+      // When formatting, only re-run SwiftLint a second time if the first
+      // invocation reported a lint failure that needs to be corrected
+      swiftLintExitCode == SwiftLintExitCode.lintFailure
+    {
+      swiftLintAutocorrectExitCode = try swiftLintAutocorrect.run()
+    } else {
+      swiftLintAutocorrectExitCode = nil
     }
 
     if
-      swiftFormat.terminationStatus == SwiftFormatExitCode.lintFailure ||
-      swiftLint.terminationStatus == SwiftLintExitCode.lintFailure
+      swiftFormatExitCode == SwiftFormatExitCode.lintFailure ||
+      swiftLintExitCode == SwiftLintExitCode.lintFailure ||
+      swiftLintAutocorrectExitCode == SwiftLintExitCode.lintFailure
```

Comment:
I ran tests locally and this is what I found as some relevant data points for the exit code when there is a lint failure.

| Tool | Lint-only command | Exit code |
| ---- | ----- | ----- |
| SwiftFormat | `swiftformat dir/with/lint/failure --config airbnb.swiftformat --lint` | 1 |
| SwiftLint | `swiftlint dir/with/lint/failure --config swiftlint.yml --strict` | 2 |

| Tool | Autocorrect command | Exit code |
| ---- | ----- | ----- |
| SwiftFormat | `swiftformat dir/with/lint/failure --config airbnb.swiftformat` | 0 |
| SwiftLint | `swiftlint dir/with/lint/failure --config swiftlint.yml --strict --fix` | 0 |

I ran these tests using this lint failure, which fails both linter and is autocorrect by both linters.
<img width="484" alt="Screenshot 2023-10-11 at 3 08 07 PM" src="https://github.com/airbnb/swift/assets/1791049/0f897dde-6542-4ec6-995c-4efc4771e061">

Reply:
This ^ behavior seems reasonable to me and seems like how our tool should function as well. What do you think about documenting the exit code behavior of this tool in the README as part of this PR and then we can ensure that the behavior matches the documented behavior. Right now AFAICT we will fail with a nonzero exit code if there is a SwiftLint error that we autocorrect, which seems undesirable.

Reply:
That seems reasonable. What do you think about making that change in a follow-up since (I think?) it's a separate behavior change from the one in this PR.

Reply:
It seems you're probably right that we throw a non-zero exit code when autocorrecting, which is probably unexpected given that the underlying tools always return a zero exit code when autocorrecting.

Reply:
https://github.com/airbnb/swift/pull/242/commits/46618a1d73d5e940fde2cbba6e008eae12464519

Reply:
> Although, what exit code would we use in a case where:
> 
> 1. We are not in lint-only mode (implying we should always return a 0 exit code?)
> 2. A SwiftLint lint-only rule triggers a failure (here that SwiftLint invocation would return a non-zero exit code)

We aligned that we _do_ want our tool to return a non-zero exit code when there is a lint-only failure even if the tool is run in autocorrect mode and the autocorrectable violations are fixed.